packages/vite/src/node/plugins/reporter.ts

```diff
@@ -111,7 +111,37 @@ export function buildReporterPlugin(config: ResolvedConfig): Plugin {
       compressedCount = 0
     },
 
-    renderChunk() {
+    renderChunk(code, chunk) {
+      for (const id of chunk.moduleIds) {
+        const module = this.getModuleInfo(id)
+        if (!module) continue
+        // When a dynamic importer shares a chunk with the imported module,
+        // warn that the dynamic imported module will not be moved to another chunk (#12850).
+        if (module.importers.length && module.dynamicImporters.length) {
+          for (const dynamicImporter of module.dynamicImporters) {
+            // Filter out the intersection of dynamic importers and sibling modules in
+            // the same chunk. The intersecting dynamic importers' dynamic import is not
+            // expected to work. Note we're only detecting the direct ineffective
+            // dynamic import here.
+            if (chunk.moduleIds.includes(dynamicImporter)) {
+              this.warn(
+                `\n(!) ${
+                  module.id
+                } is dynamically imported by ${module.dynamicImporters
+                  .map((m) => m)
+                  .join(
+                    ', ',
+                  )} but also statically imported by ${module.importers
+                  .map((m) => m)
+                  .join(
+                    ', ',
+                  )}, dynamic import will not move module into another chunk.\n`,
+              )
```

Comment:
Should we add a `break` after the `this.warn` here?

Reply:
👍🏻 Nice catch. Switched to using `.some` instead of a for loop (I start to know why it's hard to understand 😅)